File: src/main/java/com/blog/_1/dto/user/UserPublicProfileDTO.java
package com.blog._1.dto.user;

import java.util.UUID;

import lombok.Data;

@Data
public class UserPublicProfileDTO {
    private UUID id;
    private String username;
    private String bio;
    private String avatarUrl;
    private int followersCount;
    private int followingCount;
}

File: frontend/src/app/models/USER/UserPublicProfileDTO.ts
import { PostResponse } from '../POST/PostResponse';

export interface UserPublicProfileDTO {
  id: string;
  username: string;
  firstname: string;
  lastname: string;
  bio: string;
  avatarUrl: string;
  followersCount: number;
  followingCount: number;
  following?: boolean;
  posts: PostResponse[];
}
